buffer/src/main/java/io/netty/buffer/PooledDuplicatedByteBuf.java

```diff
@@ -88,13 +88,13 @@ public ByteBuf copy(int index, int length) {
     }
 
     @Override
-    public ByteBuf slice(int index, int length) {
-        return unwrap().slice(index, length);
+    public ByteBuf retainedSlice(int index, int length) {
+        return PooledSlicedByteBuf.newInstance(unwrap(), this, index, length);
     }
 
     @Override
-    public ByteBuf retainedSlice(int index, int length) {
-        return PooledSlicedByteBuf.newInstance(unwrap(), this, index, length);
+    public ByteBuf duplicate() {
+        return slice(readerIndex(), readableBytes());
```

Comment:
@Scottmitch duplicate should not return a slice... but a duplicate of the whole buffer (and not just the readable portion)